starter-core/src/main/java/io/micronaut/starter/feature/build/gradle/MicronautBuildCache.java

```diff
@@ -0,0 +1,96 @@
+package io.micronaut.starter.feature.build.gradle;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.starter.application.ApplicationType;
+import io.micronaut.starter.application.generator.GeneratorContext;
+import io.micronaut.starter.build.gradle.GradleDsl;
+import io.micronaut.starter.build.gradle.GradlePlugin;
+import io.micronaut.starter.feature.Category;
+import io.micronaut.starter.feature.Feature;
+import io.micronaut.starter.feature.build.gradle.templates.buildCache;
+import io.micronaut.starter.template.RockerWritable;
+import jakarta.inject.Singleton;
+
+@Singleton
+public class MicronautBuildCache implements Feature, BuildRemoteCacheConfiguration {
+    @Override
+    @NonNull
+    public String getName() {
+        return "micronaut-build-cache";
+    }
+
+    @Override
+    public String getTitle() {
+        return "Micronaut Build Cache";
+    }
+
+    @Override
+    @NonNull
+    public String getDescription() {
+        return "Configures Build Cache for Micronaut Gradle Enterprise";
+    }
+
+    @Override
+    public boolean supports(ApplicationType applicationType) {
+        return true;
+    }
+
+    @Override
+    public String getCategory() {
+        return Category.DEV_TOOLS;
+    }
+
+    @Override
+    public boolean isVisible() {
+        return false;
+    }
+
+    @Override
+    public String getThirdPartyDocumentation() {
+        return "https://docs.gradle.org/current/userguide/build_cache.html#sec:build_cache_configure";
+    }
+
+    @Override
+    @NonNull
+    public String getUrl() {
+        return "https://ge.micronaut.io/cache/";
+    }
+
+    @Override
+    public void apply(GeneratorContext generatorContext) {
+        if (generatorContext.getBuildTool().isGradle()) {
+            generatorContext.addBuildPlugin(GradlePlugin.builder()
+                    .settingsExtension(new RockerWritable(buildCache.template(generatorContext.getBuildTool().getGradleDsl().orElse(GradleDsl.GROOVY), this)))
+                    .build());
+        }
+    }
+
+    @Override
+    @Nullable
+    public Credentials getCredentials() {
+        String cacheUsername = System.getProperty("ge.cache.username");
```

Comment:
I am actually wondering if we should do this. It would probably make more sense that the _build script_ reads the property.

Reply:
See https://github.com/micronaut-projects/micronaut-starter/pull/1048/commits/8e03dcb8ed46edbd0bbd2b9ee85f911f294b47d6 I think that should resolve your concerns. If it does, please resolve this conversation.